app/src/main/java/com/iic/shopingo/api/Server.java

```diff
@@ -0,0 +1,101 @@
+package com.iic.shopingo.api;
+
+import android.net.Uri;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.iic.shopingo.utils.StreamToString;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.params.BasicHttpParams;
+import org.apache.http.params.HttpParams;
+import org.json.JSONObject;
+
+/**
+ * Created by assafgelber on 3/10/15.
+ */
+public class Server {
+  private static final String HEADER_AUTH_KEY = "X-WYAI-FBID";
+
+  private static final Uri BASE_URI = Uri.parse("http://iic-shopingo.herokuapp.com/");
+
+  private final HttpClient client = new DefaultHttpClient();
+
+  private String authToken;
+
+  public Server() {}
+
+  public Server(String authToken) {
+    this.authToken = authToken;
+  }
+
+  public <T> T get(String path, Class<T> responseClass) throws IOException {
+    return get(path, responseClass, new HashMap<String, Object>());
+  }
+
+  public <T> T get(String path, Class<T> responseClass, Map<String, Object> params) throws IOException {
+    Uri.Builder requestUriBuilder = BASE_URI.buildUpon().encodedPath(path);
```

Comment:
No. `path` contains the entire path.
We tried using `appendPath`, but it had issues.